src/metabase/queries/metadata.clj

```diff
@@ -1,21 +1,53 @@
 (ns metabase.queries.metadata
   (:require
    [clojure.set :as set]
+   [metabase.api.common :as api]
    [metabase.legacy-mbql.normalize :as mbql.normalize]
    [metabase.lib.schema.id :as lib.schema.id]
    [metabase.lib.util :as lib.util]
    [metabase.models.interface :as mi]
+   [metabase.permissions.core :as perms]
    [metabase.util :as u]
    [metabase.util.malli :as mu]
    [metabase.warehouse-schema.field :as schema.field]
    [metabase.warehouse-schema.table :as schema.table]
    [toucan2.core :as t2]))
 
+;; This is similar to the function in src/metabase/warehouses/api.clj, but we want to
+;; allow filtering the DBs in case of audit database, we don't want to allow users to modify
+;; its queries.
+(mu/defn- add-native-perms-info :- [:maybe
```

Comment:
Note that I could inline the audit db filter in this function, but I think it's easier to read when the predicate is put alongside the function, so it's obvious something is filtered, and we don't need to look into the function implementation to see what's being filtered.